ffi/examples/read-table/read_table.c

```diff
@@ -193,6 +194,14 @@ PartitionList* get_partition_list(SharedGlobalScanState* state)
   return list;
 }
 
+void free_partition_list(PartitionList* list) {
+  for (int i = 0; i < list->len; i++) {
+    free(list->cols+i);
```

Comment:
This looks wrong? `list->cols` presumably points to an array of pointers, and we're trying to free each of those pointers before freeing the array?
```suggestion
    free(list->cols[i]);
```

Reply:
Yep, good catch, it caused a double-free of course because it freed `list->cols+0` and then freed `list->cols` (which are the same)